node/src/components/deploy_acceptor/tests.rs

```diff
@@ -0,0 +1,584 @@
+        panic!()
+    }
+}
+
+fn put_block_to_storage(
+    block: Box<Block>,
+    responder: Responder<bool>,
+) -> impl FnOnce(EffectBuilder<Event>) -> Effects<Event> {
+    |effect_builder: EffectBuilder<Event>| {
+        effect_builder
+            .into_inner()
+            .schedule(
+                StorageRequest::PutBlock { block, responder },
+                QueueKind::Regular,
+            )
+            .ignore()
+    }
+}
+
+fn put_deploy_to_storage(
+    deploy: Box<Deploy>,
+    responder: Responder<bool>,
+) -> impl FnOnce(EffectBuilder<Event>) -> Effects<Event> {
+    |effect_builder: EffectBuilder<Event>| {
+        effect_builder
+            .into_inner()
+            .schedule(
+                StorageRequest::PutDeploy { deploy, responder },
+                QueueKind::Regular,
+            )
+            .ignore()
+    }
+}
+
+fn schedule_accept_deploy(
+    deploy: Box<Deploy>,
+    source: Source<NodeId>,
+    responder: Responder<Result<(), super::Error>>,
+) -> impl FnOnce(EffectBuilder<Event>) -> Effects<Event> {
+    |effect_builder: EffectBuilder<Event>| {
+        effect_builder
+            .into_inner()
+            .schedule(
+                super::Event::Accept {
+                    deploy,
+                    source,
+                    responder: Some(responder),
+                },
+                QueueKind::Regular,
+            )
+            .ignore()
+    }
+}
+
+async fn run_deploy_acceptor_without_timeout(
+    test_scenario: TestScenario,
+) -> Result<(), super::Error> {
+    let mut rng = crate::new_rng();
+
+    let mut runner: Runner<ConditionCheckReactor<Reactor>> =
+        Runner::new(test_scenario, &mut rng).await.unwrap();
+
+    let block = Box::new(Block::random(&mut rng));
+    // Create a responder to assert that the block was successfully injected into storage.
+    let (block_sender, block_receiver) = oneshot::channel();
+    let block_responder = Responder::create(block_sender);
+
+    runner
+        .process_injected_effects(put_block_to_storage(block, block_responder))
+        .await;
+
+    // There's only one scheduled event, so we only need to try cranking until the first time it
+    // returns `Some`.
+    while runner.try_crank(&mut rng).await.is_none() {
+        time::sleep(POLL_INTERVAL).await;
+    }
+    assert!(block_receiver.await.unwrap());
+
+    // Create a responder to assert the validity of the deploy
+    let (deploy_sender, deploy_receiver) = oneshot::channel();
+    let deploy_responder = Responder::create(deploy_sender);
+
+    // Create a deploy specific to the test scenario
+    let deploy = test_scenario.deploy(&mut rng);
+    // Mark the source as either a peer or a client depending on the scenario.
+    let source = test_scenario.source(&mut rng);
+
+    {
+        // Inject the deploy artificially into storage to simulate a previously seen deploy.
+        if test_scenario == TestScenario::FromClientRepeatedValidDeploy
```

Comment:
I would prefer keeping the "is injection needed?" logic inside the `TestScenario` implementation. So, for example, in addition to functions like `is_valid_deploy_case()` you can consider adding smth like:
```
    fn requires_injected_deploy(&self) -> bool {
        matches!(
            self,
            TestScenario::FromClientRepeatedValidDeploy | TestScenario::FromPeerRepeatedValidDeploy
        )
    }
```
and then just use `if test_scenario.requires_injected_deploy() ...`

Reply:
Applied suggestion